cpp/src/link_prediction/similarity_impl.cuh

```diff
@@ -58,7 +59,62 @@ rmm::device_uvector<weight_t> similarity(
     //        max weight((u,a), (a,v)).
     //    Use these to compute weighted score
     //
-    CUGRAPH_FAIL("weighted similarity computations are not supported in this release");
+
+    rmm::device_uvector<weight_t> similarity_score(num_vertex_pairs, handle.get_stream());
+
+    //
+    //  Compute vertex_degree for all vertices, then distribute to each GPU.
+    //  Need to use this instead of the dummy properties below
+    //
+
+    rmm::device_uvector<weight_t> weighted_out_degrees =
+      compute_out_weight_sums(handle, graph_view, *edge_weight_view);
+
+    rmm::device_uvector<weight_t> vertex_weights =
+      compute_out_weight_sums(handle, graph_view, *edge_weight_view);
+
+    per_v_pair_transform_dst_nbr_intersection(
+      handle,
+      graph_view,
+      *edge_weight_view,
+      vertex_pairs_begin,
+      vertex_pairs_begin + num_vertex_pairs,
+      weighted_out_degrees.begin(),
+      [functor] __device__(auto a,
+                           auto b,
+                           auto weight_a,
+                           auto weight_b,
+                           auto intersection,
+                           auto intersected_properties_a,
+                           auto intersected_properties_b) {
+        weight_t min_weight_a_intersect_b = weight_t{0};
+        weight_t max_weight_a_intersect_b = weight_t{0};
```

Comment:
This can be `std::numeric_limits<weight_t>::lowest()` if intersection properties can have a negative number. Or if you're expecting the property (I assume edge weights) to be strictly non-negative, this should be checked if `do_expensive_check` flag is set.